### PR TITLE
re-arranged parameter to keep binary back compat

### PR DESCRIPTION
**Pull request by heejaechang**
fix https://github.com/dotnet/roslyn/issues/9047

### CODE REVIEW

**Comment by heejaechang** [comment]
@jaredpar @mavasani @srivatsn can you take a look?

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -11,6 +11,8 @@ Microsoft.CodeAnalysis.Diagnostics.CompilationStartAnalysisContext.TryGetValue<T
...
+Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, bool concurrentAnalysis, bool logAnalyzerExecutionTime) -> void
+Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, bool concurrentAnalysis, bool logAnalyzerExecutionTime, bool reportSuppressedDiagnostics) -> void
```
Comment (mavasani) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: Why are we adding 2 new public APIs here? I would recommend adding just one, with `analyzerFilter` being the last parameter in that. So we will have following 3 public APIs (first 2 being the ones already shipped in Update1):

```c#
Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, bool concurrentAnalysis, bool logAnalyzerExecutionTime) -> void

Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, bool concurrentAnalysis, bool logAnalyzerExecutionTime, bool reportSuppressedDiagnostics) -> void

Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, bool concurrentAnalysis, bool logAnalyzerExecutionTime, bool reportSuppressedDiagnostics, System.Func<System.Exception, bool> analyzerExceptionFilter) -> void
```

Comment (heejaechang) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: sure.

**Comment by heejaechang** [comment]
@davkean @jaredpar I will port it to stabilization branch asking ask mode stuff.

**Comment by heejaechang** [comment]
ping? can I get sign off. once I have sign off, I will port it to stabilization and send email to matt for ask mode permission.

**Comment by davkean** [comment]
Just retarget the PR to stabilization. We'll automatically merge into master.

**Pull request closed without merging**